Add global and component level network configuration options

timgerstel opened the pull request:
<!-- Thank you for your pull request! Please provide a description of the changes in this PR in the field above and provide the following information. -->

Please check if your PR fulfills the following requirements. This is simply a reminder of what we are going to look for before merging your PR. If you don't know all of this information when you create this PR, don't worry. You can edit this template as you're working on it.

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Necessary documentation (if appropriate) have been added / updated
- [ ] DCO signoffs have been added to all commits, including this PR

#### PR type
What type of changes does your PR introduce to Zowe? _Put an `x` in the box that applies to this PR. If you're unsure about any of them, don't hesitate to ask._ 
- [ ] Bugfix <!-- non-breaking change which fixes an issue -->
- [ ] Feature <!-- non-breaking change which adds functionality -->
- [ ] Other... Please describe:

#### Relevant issues
<!-- Link to a relevant GitHub issue if any. If this PR applies to multiple issues, preface each issue reference with the "Fixes" keyword. For example, Fixes #123, Fixes #456. -->

Fixes <!-- Issue number -->

#### Changes proposed in this PR
<!-- Please describe your Pull Request. -->
- 

#### Does this PR introduce a breaking change?
<!-- Is this a fix or feature that would cause existing functionality to not work as expected? -->

- [ ] Yes
- [ ] No

<!-- If yes, please describe the impact and migration path below.-->

#### Does this PR do something the person installing Zowe should know about?

<!-- Is this a fix or a feature that changes customizable files (e.g. configuration files, sample JCL, etc),
product requirements, or other installation or configuration related area? If so, please fill out the template below. 
There is no need to report the need to restart the servers to activate the change. Note that the provided text will be formatted in 64-character lines, and that round brackets, ( and ), must always be used in matching pairs. -->

---
* Affected function: _general area of interest_ *

---
* Description: _1 line description_ *

---
* Part: _name of customizable file involved_  *

---
_multi-line description_

#### Is there a related doc issue or Pull Request? 
<!-- Link to a relevant documentation issue or Pull Request if any. -->

Doc issue/PR number: 

#### Other information

timgerstel committed:
Add global and component level network configuration options

Signed-off-by: Timothy Gerstel <tim.gerstel@gmail.com>

1000TurquoisePogs commented on schemas/zowe-yaml-schema.json:
```diff
@@ -423,6 +423,35 @@
+            },
+            "maxTls": {
+              "type": "string",
+              "description": "Maximum TLS version allowed for network connections. Must be: 1.0, 1.1, 1.2, or 1.3."
```
if it 'must be' anything, make it an enum.
"enum": [ "1.0", "1.1" ]...
I agree with enum because "string" is unhelpful, it doesnt tell you what's possible.

1000TurquoisePogs commented on schemas/zowe-yaml-schema.json:
```diff
@@ -423,6 +423,35 @@
+              "type": "array",
+              "description": "The IP addresses which all of the Zowe servers will be binding to.",
+              "items": {
+                "type": "string"
```
dont make it a string. cant do good validation on that.
use this: https://github.com/zowe/zowe-install-packaging/blob/v2.x/staging/schemas/server-common.json#L78

and we should have an ipv6 one too, even though we dont support ipv6 yet...

timgerstel committed:
Use zoweIpv4 for bind addresses and update tls settings to enums

Signed-off-by: Timothy Gerstel <tim.gerstel@gmail.com>

timgerstel committed:
Merge branch 'v2.x/staging' into schema/network/https

JoeNemo reviewed (approved):
Approved.

timgerstel committed:
Merge branch 'v2.x/staging' into schema/network/https

1000TurquoisePogs committed:
Deduplicate schema and separate server from client

Also change "bind" to "listen" and tls prefix with TLSv, as seen in the majority of our servers.

Signed-off-by: 1000TurquoisePogs <sgrady@rocketsoftware.com>

1000TurquoisePogs committed:
Fix schema json syntax

Signed-off-by: 1000TurquoisePogs <sgrady@rocketsoftware.com>

1000TurquoisePogs committed:
Shorten tlsSettings to tls, and fix plural ipAddresses

Signed-off-by: 1000TurquoisePogs <sgrady@rocketsoftware.com>

1000TurquoisePogs committed:
Merge branch 'v2.x/staging' into schema/network/https

1000TurquoisePogs commented:
Tested with https://github.com/zowe/zowe-install-packaging/pull/3638 and reviewed by @sakshibobade21 too.
